### ISSUE TITLE
Small Change to  yii.gridView.js please 

### ISSUE THREAD

**Issue by stovesy** [yiisoft/yii2#20064]
Would it be possible to alter the above as follows please.

 var event = $.Event(gridEvents.beforeFilter);
  $grid.trigger(event, $form);
  if (event.result === false) {
      return;
  }

  $form.submit();

  $grid.trigger(gridEvents.afterFilter, $form);

Passing the $form object to the events will allow me to access and add additional info for the filter processing

**Comment by samdark** [comment]
Yes. Good idea, I think. Do you have some time for a pull request?

**Comment by xicond** [comment]
I tought, $form already inside $grid, can just acquire by $(this).children('form.gridview-filter-form')
